insalan/tournament/views.py

```diff
@@ -164,6 +164,13 @@ def get(self, request, primary_key: int):
                 tourney.game, context={"request": request}
             ).data
 
+            user_player = None
+            if request.user.is_authenticated:
+                try:
+                    user_player = Player.objects.get(user=request.user)
```

Comment:
This will not work for two reasons :
- Player.objects.get will also get the older registrations.
- As you are checking Player, you will not have payment status if registration is Manager or Substitute
To fix the first problem, you can add the tournament id to the Player query. For the second, you can use union betwen player, manager and substitutes.